hudi-common/src/main/java/org/apache/hudi/common/util/ObjectSizeCalculator.java

```diff
@@ -90,7 +90,7 @@ public static long getObjectSize(Object obj) throws UnsupportedOperationExceptio
   private final Map<Class<?>, ClassSizeInfo> classSizeInfos = new IdentityHashMap<>();
 
   private final Set<Object> alreadyVisited = Collections.newSetFromMap(new IdentityHashMap<>());
-  private final Deque<Object> pending = new ArrayDeque<>(16 * 1024);
+  private final Deque<Object> pending = new ArrayDeque<>(64);
```

Comment:
Can you explain this change?

Reply:
Reducing upfront allocation w/ every Calculator object from 128kb to 512b